Action doesn't fail on compiler errors

For example: (https://github.com/zjeffer/Waybar/actions/runs/7129252582/job/19412930401)

```
  INFO:CPP Linter:Running "clang-tidy-17 --export-fixes=.cpp-linter_cache/clang_tidy_output.yml -p /__w/Waybar/Waybar/build --line-filter=[{"name": "src/modules/hyprland/workspaces.cpp", "lines": [[50, 51]]}] --extra-arg= src/modules/hyprland/workspaces.cpp"
  DEBUG:CPP Linter:Output from clang-tidy:
  ../include/bar.hpp:16:10: error: 'xdg-output-unstable-v1-client-protocol.h' file not found [clang-diagnostic-error]
     16 | #include "xdg-output-unstable-v1-client-protocol.h"
        |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  
  DEBUG:CPP Linter:clang-tidy made the following summary:
  13697 warnings and 2 errors generated.
  Error while processing /__w/Waybar/Waybar/src/modules/hyprland/workspaces.cpp.
  Suppressed 13697 warnings (13670 in non-user code, 17 due to line filter, 10 with check filters).
  Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
  Found compiler error(s).
```

```
Posting comment(s)
  INFO:CPP Linter:0 checks-failed
```



Relevant parts of my workflow file:
```
      with:
        style: '' # empty string => don't do clang-format checks here, we do them in clang-format.yml
        files-changed-only: true # only check files that have changed
        lines-changed-only: true # only check lines that have changed
        tidy-checks: '' # empty string => use the .clang-tidy file
        version: '17' # clang-tools version
        database: 'build' # path to the compile_commands.json file
    - name: Check if clang-tidy failed on any files
      if: steps.clang-tidy-check.outputs.checks-failed > 0
      run: echo "Some files failed the linting checks!" && exit 1
```
checks-failed is 0, because the compiler error apparently doesn't count as a failed check.

How can I ensure the action fails when there are compiler errors like missing header files?

--- a/cpp_linter/clang_tidy.py
+++ b/cpp_linter/clang_tidy.py
@@ -42,7 +42,7 @@ class TidyNotification:
         self.cols = int(self.cols)
         #: The source filename concerning the notification.
         self.filename = (
-            PurePath(self.filename).as_posix().replace(Path.cwd().as_posix(), "")
+            Path(self.filename).resolve().as_posix().replace(Path.cwd().as_posix(), "")
         )
         #: A `list` of lines for the code-block in the notification.
         self.fixit_lines: List[str] = []

--- a/cpp_linter/clang_tidy.py
+++ b/cpp_linter/clang_tidy.py
@@ -42,7 +42,10 @@ class TidyNotification:
         self.cols = int(self.cols)
         #: The source filename concerning the notification.
         self.filename = (
-            Path(self.filename).resolve().as_posix().replace(Path.cwd().as_posix(), "")
+            Path(self.filename)
+            .resolve()
+            .as_posix()
+            .replace(Path.cwd().as_posix() + "/", "")
         )
         #: A `list` of lines for the code-block in the notification.
         self.fixit_lines: List[str] = []

--- a/cpp_linter/run.py
+++ b/cpp_linter/run.py
@@ -604,17 +604,16 @@ def make_annotations(
                 Globals.format_failed_count += 1
     for note in GlobalParser.tidy_notes:
         if lines_changed_only:
-            filename = note.filename.replace("\\", "/").lstrip("..").lstrip("/")
             line_filter = []
             for file in files:
-                if filename == file["filename"]:
+                if note.filename == file["filename"]:
                     line_filter = cast(
                         List[int], range_of_changed_lines(file, lines_changed_only)
                     )
                     break
             else:  # pragma: no cover
                 # filename match not found; treat as error from non-user code
-                logger.debug("%s != %s", filename, file["filename"])
+                logger.debug("%s doesn't match any project files", note.filename)
                 continue
             if note.line in line_filter or not line_filter:
                 Globals.tidy_failed_count += 1

--- a/cpp_linter/run.py
+++ b/cpp_linter/run.py
@@ -607,13 +607,13 @@ def make_annotations(
             filename = note.filename.replace("\\", "/").lstrip("..").lstrip("/")
             line_filter = []
             for file in files:
-                print(filename, "?=", file["filename"])
                 if filename == file["filename"]:
                     line_filter = cast(
                         List[int], range_of_changed_lines(file, lines_changed_only)
                     )
                     break
             else:  # filename match not found; treat line_filter as empty list
+                logger.debug("%s != %s", filename, file["filename"])
                 continue
             if note.line in line_filter or not line_filter:
                 Globals.tidy_failed_count += 1

--- a/cpp_linter/run.py
+++ b/cpp_linter/run.py
@@ -604,7 +604,7 @@ def make_annotations(
                 Globals.format_failed_count += 1
     for note in GlobalParser.tidy_notes:
         if lines_changed_only:
-            filename = note.filename.replace("\\", "/").lstrip("/")
+            filename = note.filename.replace("\\", "/").lstrip("..").lstrip("/")
             line_filter = []
             for file in files:
                 print(filename, "?=", file["filename"])

--- a/cpp_linter/run.py
+++ b/cpp_linter/run.py
@@ -612,7 +612,8 @@ def make_annotations(
                         List[int], range_of_changed_lines(file, lines_changed_only)
                     )
                     break
-            else:  # filename match not found; treat line_filter as empty list
+            else:  # pragma: no cover
+                # filename match not found; treat as error from non-user code
                 logger.debug("%s != %s", filename, file["filename"])
                 continue
             if note.line in line_filter or not line_filter:

--- a/tests/capture_tools_output/test_tools_output.py
+++ b/tests/capture_tools_output/test_tools_output.py
@@ -283,14 +283,14 @@ def test_tidy_annotations(
     make_annotations(
         style="", file_annotations=True, lines_changed_only=lines_changed_only
     )
-    for message in [r.message for r in caplog.records if r.levelno == logging.INFO]:
+    messages = [r.message for r in caplog.records if r.levelno == logging.INFO]
+    assert messages
+    for message in messages:
         if TIDY_RECORD.search(message) is not None:
             line = int(TIDY_RECORD_LINE.sub("\\1", message))
-            file_obj = match_file_json(
-                RECORD_FILE.sub("\\1", message).replace("\\", "/")
-            )
-            if file_obj is None:
-                continue
+            filename = RECORD_FILE.sub("\\1", message).replace("\\", "/")
+            file_obj = match_file_json(filename)
+            assert file_obj is not None, f"{filename} was not matched with project src"
             ranges = cpp_linter.range_of_changed_lines(file_obj, lines_changed_only)
             if ranges:  # an empty list if lines_changed_only == 0
                 assert line in ranges

